include/zephyr/devicetree.h

```diff
@@ -4418,6 +4432,18 @@
 #define DT_INST_STRING_TOKEN_BY_IDX(inst, prop, idx) \
 	DT_STRING_TOKEN_BY_IDX(DT_DRV_INST(inst), prop, idx)
 
+/**
+ * @brief Like DT_INST_STRING_TOKEN_BY_IDX(), but with a fallback to default_value
+ * @param inst instance number
+ * @param prop lowercase-and-underscores property name
+ * @param idx the index to get
+ * @param default_value a fallback value to expand to
+ * @return the element in @p prop at index @p idx as a token if indexed property
+ *         is present, @p default_value otherwise
```

Comment:
Following existing conventions:
```suggestion
 * @return the element in @p prop at index @p idx as a token, or @p default_value
```